include/oneapi/dpl/internal/binary_search_impl.h

```diff
@@ -83,43 +83,46 @@ struct custom_brick
 template <class _Tag, typename Policy, typename InputIterator1, typename InputIterator2, typename OutputIterator,
           typename StrictWeakOrdering>
 OutputIterator
-lower_bound_impl(_Tag, Policy&& policy, InputIterator1 start, InputIterator1 end, InputIterator2 value_start,
+lower_bound_impl(_Tag tag, Policy&& policy, InputIterator1 start, InputIterator1 end, InputIterator2 value_start,
                  InputIterator2 value_end, OutputIterator result, StrictWeakOrdering comp)
 {
     static_assert(__internal::__is_host_dispatch_tag_v<_Tag>);
 
-    return oneapi::dpl::transform(policy, value_start, value_end, result,
-                                  [=](typename ::std::iterator_traits<InputIterator2>::reference val) {
-                                      return ::std::lower_bound(start, end, val, comp) - start;
-                                  });
+    return oneapi::dpl::__internal::__pattern_walk2(
+        tag, std::forward<Policy>(policy), value_start, value_end, result,
+        oneapi::dpl::__internal::__transform_functor{[=](typename std::iterator_traits<InputIterator2>::reference val) {
```

Comment:
As far as we are changing this place... Lets changes 'val' type as well, according to 
https://en.cppreference.com/w/cpp/algorithm/lower_bound, to `const T& value`
Also it is relevant to `upper_bound_impl`

Reply:
Done